sw/device/lib/dif/dif_otbn.h

```diff
@@ -81,8 +81,20 @@ typedef enum dif_otbn_err_code {
   kDifOtbnErrCodeNoError = 0x0,
   /** Load or store to invalid address. */
   kDifOtbnErrCodeBadDataAddr = 0x1,
+  /** Instruction fetch from invalid address. */
+  kDifOtbnErrCodeBadInsnAddr = 0x2,
   /** Call stack underflow or overflow. */
-  kDifOtbnErrCodeCallStack = 0x2,
+  kDifOtbnErrCodeCallStack = 0x3,
+  /** Illegal instruction execution attempted */
+  kDifOtbnErrCodeIllegalInsn = 0x4,
+  /** LOOP[I] related error */
+  kDifOtbnErrCodeLoop = 0x5,
+  /** ECC error seen in Imem read */
+  kDifOtbnErrCodeFatalImemECC = 0x80,
+  /** ECC error seen in Dmem read */
+  kDifOtbnErrCodeFatalDmemECC = 0x81,
+  /** ECC error seen in RF */
+  kDifOtbnErrCodeFatalRegECC = 0x82
```

Comment:
Why are these error codes, which need to be kept in sync with HW, exposed in the header? This seems like something that should have gone in the .c file.

Reply:
We are looking at autogenerating these but otherwise we could alter the DIF to map between hardware error codes and DIF enum error codes.

Reply:
I'm happy with these being in the header - it's not like a different parameterisation of the DIF means there's a different set of error codes. If anything, it is a follow-up issue anyway.